### PR TITLE
Pine - Michelle B.

### PR DESCRIPTION
**Pull request by michellebodart**

### CODE REVIEW

File: swap_meet/clothing.py [hunk 1]
```diff
@@ -1,2 +1,14 @@
...
+class Clothing(Item):
+    
+    def __init__(self, category="Clothing", condition = 0, age = 0):
+        super().__init__(category, condition, age)
+
+    def __str__(self):
+        return "The finest clothing you could wear."
+
+    def condition_description(self):
+        return super().condition_description()
```
Comment (CheezItMan) [inline swap_meet/clothing.py]: This is actually unnecessary since it does the same thing as the `Item` method.

```suggestion
```

File: swap_meet/decor.py [hunk 2]
```diff
@@ -1,2 +1,12 @@
-class Decor:
-    pass
+from swap_meet.item import Item
+
+class Decor(Item):
```
Comment (CheezItMan) [inline swap_meet/decor.py:3]: 👍

File: swap_meet/decor.py [hunk 3]
```diff
@@ -1,2 +1,12 @@
...
+    
+    def __init__(self, category = "Decor", condition = 0, age = 0):
+        super().__init__(category, condition, age)
+    
+    def __str__(self):
+        return "Something to decorate your space."
+
+    def condition_description(self):
+        return super().condition_description()
```
Comment (CheezItMan) [inline swap_meet/decor.py]: Unneeded

```suggestion
```

File: swap_meet/electronics.py [hunk 4]
```diff
@@ -1,2 +1,12 @@
...
+class Electronics(Item):
+    
+    def __init__(self, category = "Electronics", condition = 0, age = 0):
+        super().__init__(category, condition, age)
+
+    def __str__(self):
+        return "A gadget full of buttons and secrets."
+
+    def condition_description(self):
+        return super().condition_description()
```
Comment (CheezItMan) [inline swap_meet/electronics.py]:
```suggestion
```

File: swap_meet/electronics.py [hunk 5]
```diff
@@ -1,2 +1,12 @@
-class Electronics:
-    pass
+from swap_meet.item import Item
+
+class Electronics(Item):
```
Comment (CheezItMan) [inline swap_meet/electronics.py:3]: 👍

File: swap_meet/item.py [hunk 6]
```diff
@@ -1,2 +1,22 @@
+CONDITIONS_DICTIONARY = {
+    0: "terrible",
+    1: "pretty bad",
+    2: "not great",
+    3: "meh",
+    4: "nice",
+    5: "brand new",
+}
```
Comment (CheezItMan) [inline swap_meet/item.py:8]: Love this dictionary!

File: swap_meet/item.py [hunk 7]
```diff
@@ -1,2 +1,22 @@
...
+
 class Item:
```
Comment (CheezItMan) [inline swap_meet/item.py:10]: 👍

File: swap_meet/testing_file.py [hunk 8]
```diff
@@ -0,0 +1,48 @@
+from vendor import Vendor
```
Comment (CheezItMan) [inline swap_meet/testing_file.py:1]: This file doesn't need to be committed

File: swap_meet/vendor.py [hunk 9]
```diff
@@ -1,2 +1,75 @@
 class Vendor:
-    pass
+    
+    def __init__(self, inventory=None):
+        if not inventory:
+            self.inventory = []
+        else:
+            self.inventory = inventory
```
Comment (CheezItMan) [inline swap_meet/vendor.py:7]: 👍 Good default parameter

File: swap_meet/vendor.py [hunk 10]
```diff
@@ -1,2 +1,75 @@
...
+
+    def add(self, new_item):
+        self.inventory.append(new_item)
+        return new_item
+
+    def remove(self, item_to_remove):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:13]: 👍

File: swap_meet/vendor.py [hunk 11]
```diff
@@ -1,2 +1,75 @@
...
+        if item_to_remove in self.inventory:
+            self.inventory.remove(item_to_remove)
+            return item_to_remove
+        else:
+            return False
+
+    def get_by_category(self, category):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:20]: 👍

File: swap_meet/vendor.py [hunk 12]
```diff
@@ -1,2 +1,75 @@
...
+        result = []
+        for thing in self.inventory:
+            if thing.category == category:
+                result.append(thing)
+        return result
+
+    def swap_items(self, other_vendor, self_item, other_vendor_item):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:27]: 👍

File: swap_meet/vendor.py [hunk 13]
```diff
@@ -1,2 +1,75 @@
...
+        if self_item in self.inventory and other_vendor_item in other_vendor.inventory:
+            self.add(other_vendor_item)
+            other_vendor.remove(other_vendor_item)
+            other_vendor.add(self_item)
+            self.remove(self_item)
+            return True
+        else:
+            return False
+    
+    def swap_first_item(self, other_vendor):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:37]: 👍 Good reuse of existing methods

File: swap_meet/vendor.py [hunk 14]
```diff
@@ -1,2 +1,75 @@
...
+        if len(self.inventory) >= 1 and len(other_vendor.inventory) >= 1:
+            return self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
+        else:
+            return False
+
+    def get_best_by_category(self, selected_category):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:43]: 👍 Good reuse of existing methods

File: swap_meet/vendor.py [hunk 15]
```diff
@@ -1,2 +1,75 @@
...
+        result = None
+        item_list = self.get_by_category(selected_category)
+        best_score = 0
+        for item in item_list:
+            if item.condition >= best_score:
+                best_score = item.condition
+                result = item
+        return result
+
+    def swap_best_by_category(self, other = None, my_priority = "", their_priority = ""):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:53]: 👍 Good reuse of existing methods

File: swap_meet/vendor.py [hunk 16]
```diff
@@ -1,2 +1,75 @@
...
+        my_item = self.get_best_by_category(their_priority)
+        their_item = other.get_best_by_category(my_priority)
+        if not ( my_item and their_item ):
+            return False
+        else:
+            return self.swap_items(other,my_item, their_item)
+
+    def swap_by_newest(self, other = None):
```
Comment (CheezItMan) [inline swap_meet/vendor.py]: You could make a helper method to get newest and then call your swap methods.

File: tests/test_wave_07.py [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+    item_c = Clothing(condition=4.0, age = 100)
+    item_d = Decor(condition=5.0, age = 7)
+    item_e = Clothing(condition=3.0, age = 3)
+    item_f = Electronics(condition = 0.0, age = 98)
+    devin = Vendor(inventory=[item_a, item_b, item_c,])
+    renee = Vendor(inventory = [item_d, item_e, item_f])
+
+    result = devin.swap_by_newest(renee)
+
+    assert result
```
Comment (CheezItMan) [inline tests/test_wave_07.py:19]: You should also verify that `devin` and `renee` have swapped items.

File: tests/test_wave_07.py [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+
+def test_swap_by_newest_empty_inventory_is_false():
+    devin = Vendor()
+    renee = Vendor()
+
+    result = devin.swap_by_newest(renee)
+
+    assert not result
```
Comment (CheezItMan) [inline tests/test_wave_07.py:33]: This is a great edge-case test, I would also write a test with one vendor with empty inventory and the other that has inventory.

File: tests/test_wave_07.py [hunk 19]
```diff
@@ -0,0 +1,49 @@
+import pytest
```
Comment (CheezItMan) [inline tests/test_wave_07.py:1]: Yay! tests!

**Review by CheezItMan — COMMENTED** [review]
Nice work with granular, meaningful commits.  You got everything done *quickly* and even did the extras!  You made me really happy with the written tests.  Well done.  🕺
